### DISCUSSION TITLE
Error while running reinstall.sh on mac with Apple Silicon

### DISCUSSION THREAD

**Discussion by Pedrohmlara** [NVIDIA-NeMo/NeMo#7069] [Q&A]
I'm trying to install NeMo on Mac with M2, but when I run 
`./reinstall.sh` I got the following error:

`ERROR: Cannot uninstall 'llvmlite'. It is a distutils installed project and thus we cannot accurately determine which files belong to it which would lead to only a partial uninstall.`

Any idea to fix this?
Thanks!

**Comment by titu1994** [comment]
You need to install Numba via conda or pip or some other method. Llvmlite is part of that library

**Reply by Pedrohmlara** [reply]
I ran this `conda install -c conda-forge numba` in the same conda env and installed. But when I run `./reinstall.sh` again I got the same error message....

![Screenshot 2023-07-19 at 09 17 30](https://github.com/NVIDIA/NeMo/assets/41451816/250cdd19-7f3d-4000-a7b6-1391edc52e4e)

**Reply by titu1994** [reply]
Hmm odd - just to confirm the conda env was active and the default for "python XYZ" ? Cause if conda already installed it then it should not show up in the list. Maybe related to this - https://github.com/numba/llvmlite/issues/693 ?

**Reply by Pedrohmlara** [reply]
Well, the conda env is activated with the right python version.

I've tried running the installation commands for llvmlite manually, it works, but when I run this .sh the error is repeated. I also tried uninstalling and running .sh but it still doesn't work. (I looked at what you suggested linked here in the thread but I couldn't figure it out...)

**Reply by titu1994** [reply]
Hmm so pip install .[all] works from the source branch but reinstall.sh doesn't work ?

**Reply by Pedrohmlara** [reply]
Exactly...
And because of this I can't proceed with the installation of NeMo...

Thanks so much for this support!

**Reply by titu1994** [reply]
But the recommended way to install Nemo is to use pip install anyway ? I don't see the need to use reinstall.sh if it's failing for you but pip is working. Both achieve the exact same thing.
